### PR TITLE
Remove ENABLE_REMIX_ROUTER flags

### PR DESCRIPTION
**Pull request by brophdawg11**
Removes the no-longer used `ENABLE_REMIX_ROUTER` flag

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -23,10 +23,3 @@ jobs:
     uses: ./.github/workflows/reusable-test.yml
     with:
       node_version: '["latest"]'
-
-  test-experimental:
-    if: github.repository == 'remix-run/remix'
-    uses: ./.github/workflows/reusable-test.yml
-    with:
-      node_version: '["latest"]'
-      enable_remix_router: "1"
```
Comment (brophdawg11) [inline .github/workflows/test.yml:32]: No more dup test runs 🥳

File: packages/remix-server-runtime/rollup.config.js [hunk 2]
```diff
@@ -14,15 +13,6 @@ const {
...
 const { name: packageName, version } = require("./package.json");
 
-const ENABLE_REMIX_ROUTER = !!process.env.ENABLE_REMIX_ROUTER;
-
-const replacePlugin = replace({
-  preventAssignment: true,
-  values: {
-    "process.env.ENABLE_REMIX_ROUTER": ENABLE_REMIX_ROUTER ? "1" : "0",
-  },
-});
```
Comment (brophdawg11) [inline packages/remix-server-runtime/rollup.config.js:24]: Revert the changes from https://github.com/remix-run/remix/pull/4245

**Review by MichaelDeBoey — APPROVED** [review]
